### PR TITLE
Add more in-depth and user-facing geometry tutorial

### PR DESCRIPTION
**Pull request by CoronelBuendia**
## Description
Adds a  more in-depth user tutorial for geometry.

## Interface Changes

N/A
## Checklist

I confirm that I have completed the following checks:

- [x] Tests run locally and pass `pytest tests --reactor`
- [x] Code quality checks run locally and pass `flake8` and `black .`
- [x] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

**Comment by je-cook** [comment]
I thought we were doing python files that are auto converted to ipynb?

File: examples/geometry/geometry_tutorial.py [hunk 1]
```diff
@@ -20,141 +20,297 @@
...
+for i, shell in enumerate(cylinder.boundary):
+    print(f"Shell: {i}.{j}.{k} is a BluemiraShell: {isinstance(shell, BluemiraShell)}")
+    for j, face in enumerate(shell.boundary):
+        print(f"Face: {i}.{j}.{k} is a BluemiraFace: {isinstance(face, BluemiraFace)}")
+        for k, wire in enumerate(face.boundary):
+            print(
+                f"Wire: {i}.{j}.{k} is a BluemiraWire: {isinstance(wire, BluemiraWire)}"
+            )
+
+# OK, so a cylinder is pretty simple, but more complicated shapes
```
Comment (je-cook) [inline examples/geometry/geometry_tutorial.py:142]: Markdown these lines?

File: examples/geometry/geometry_tutorial.py [hunk 2]
```diff
@@ -20,141 +20,297 @@
...
+plot_2d(pentagon)
+
+# %%
+# Polygons are good for things with straight lines.
+# Circles you've met already.
+# For everything else, there's splines.
+
+# Say you have a weird shape, that you might calculate via a equation.
+# It's not a good idea to make a polygon with lots of very small sides
+# for this. It's computationally expensive, and it will look ugly.
```
Comment (je-cook) [inline examples/geometry/geometry_tutorial.py:177]: Markdown here

File: examples/geometry/geometry_tutorial.py [hunk 3]
```diff
@@ -20,141 +20,297 @@
...
+# %%
+# Sweep a profile along a path
+
+points = np.array([[4.5, 4.5], [0, 3], [2.5, 2.5]]).T
+straight_line = make_polygon(points)
+quarter_turn = make_circle(center=(3, 3, 2.5), axis=(0, 0, 1), radius=1.5, end_angle=90)
+path = BluemiraWire([straight_line, quarter_turn])
+solid = sweep_shape(rectangle.boundary[0], path)
+show_cad(solid)
+# %%[markdown]
```
Comment (je-cook) [inline examples/geometry/geometry_tutorial.py:247]: space needed for markdown to render

File: examples/geometry/geometry_tutorial.py [hunk 4]
```diff
@@ -20,141 +20,297 @@
...
+box_2 = extrude_shape(box_2, (0, 1, 0))
+
+fused_boxes = boolean_fuse([box_1, box_2])
+
+show_cad(fused_boxes)
+
+cut_box_1 = boolean_cut(box_1, box_2)[0]
+
+show_cad(cut_box_1)
+# %%[markdown]
```
Comment (je-cook) [inline examples/geometry/geometry_tutorial.py:285]: newline for markdown

File: examples/geometry/geometry_tutorial.py [hunk 5]
```diff
@@ -20,141 +20,297 @@
...
+new_cut_box_1.scale(3)
+blue_red_options = [DisplayCADOptions(color="blue"), DisplayCADOptions(color="red")]
+show_cad([cut_box_1, new_cut_box_1], options=blue_red_options)
+
+# %%[markdown]
+# ## Exporting geometry
+
+# At present, only the STEP Assembly format is supported
+# for exporting geometry.
+# %%
```
Comment (je-cook) [inline examples/geometry/geometry_tutorial.py:309]: Markdown conversion hasnt worked here

**Review by je-cook — CHANGES_REQUESTED** [review]
Generally looks good. Do we want two geometry tutorials (this as maybe a user one)? Otherwise there are conflicts with the tutorial Dan put up a few days ago.

**Comment by CoronelBuendia** [comment]
> Generally looks good. Do we want two geometry tutorials (this as maybe a user one)? Otherwise there are conflicts with the tutorial Dan put up a few days ago.

Yeah I meant to tell Dan I was going to write a new one. This is intended to replace the one we had, which had less of a user focus. (e.g. I doubt people will making solids directly from vertices).

**Review by je-cook — APPROVED** [review]
Ok  fair enough, merging

**Pull request merged**